checklock_holmes/checklock_holmes/nb_checker.py

```diff
@@ -0,0 +1,176 @@
+
+from checklock_holmes.utils.helpers import CuriousIncident
+
+_DEFAULT_ENV_VARIABLES = {
+    'OBJECTIV_VERSION_CHECK_DISABLE': 'true'
+}
+_IGNORE_MARKDOWN_SQL_DISPLAYS = True
+
+
+@dataclass
+class NoteBookChecker:
+    MAX_LOG_EXCEPTION_MESSAGE = 500
+
+    metadata: NoteBookMetadata
+    display_cell_timing: bool
+
+    _errors: Dict[SupportedEngine, CellError] = field(default_factory=dict, init=False)
+    _cell_timings: List[CellTiming] = field(default_factory=list, init=False)
+
+    @cached_property
+    def cells(self) -> List[Dict[str, Any]]:
+        with open(self.metadata.path) as nb_file:
+            nb_data = json.load(nb_file)
+            return nb_data.get('cells')
+
+    def check_notebook(self, engine: SupportedEngine) -> NoteBookCheck:
+        """
+        Creates and executes the notebook's script for the provided engine.
+
+        Returns a generated report based on the execution.
+        """
+        self._cell_timings = []
+        wrapped_script = self.get_script(engine, is_execution=True)
+        completed = True
+
+        start_time = time.time()
+        try:
+            exec(wrapped_script)
+        except Exception as exc:
+            if not self._errors.get(engine):
+                raise CuriousIncident(notebook_name=self.metadata.name, exc=exc)
+
+            completed = False
+        end_time = time.time()
+
+        error = self._errors.get(engine)
+        return NoteBookCheck(
+            metadata=self.metadata,
+            engine=engine,
+            completed=completed,
+            error=error,
+            failing_block=''.join(self.cells[error.number]['source']) if error else None,
+            elapsed_time=end_time - start_time,
+            elapsed_time_per_cell=self._cell_timings,
+        )
+
+    @staticmethod
+    def _time_wrapped_cel_code(cell_number: int, code: str) -> str:
+        """
+        Wraps the cell's code for tracking the elapsed time on execution.
+        """
+        return TIMING_CELL_CODE_TEMPLATE.format(
+            code_to_time=code,
+            timing_stmt=f'self._log_cell_timing({cell_number}, elapsed_time)'
+        )
+
+    def _log_wrapped_cell_code(self, cell_number: int, engine: str, source: List[str]) -> str:
+        """
+        Wraps the cell's code for tracking raised exceptions on execution.
+        """
+        code = WRAPPED_CODE_TEMPLATE.format(
+            code_to_wrap='    '.join(source),
+            error_log_stmt=f'self._log_error({cell_number}, \"{engine}\",  e)',
+        )
+        if self.display_cell_timing:
+            code = self._time_wrapped_cel_code(cell_number, code)
+
+        return code
+
+    def _log_error(self, cell_number: int, engine: SupportedEngine,  exc: Exception):
+        """
+        Logs raised exception when running the cell.
+        """
+        self._errors[engine] = CellError(
+            number=cell_number,
+            exc=f'{exc.__class__.__name__}: {exc.args[0][:self.MAX_LOG_EXCEPTION_MESSAGE]}...'
```

Comment:
```suggestion
            exc=f'{exc.__class__.__name__}: {str(exc)[:self.MAX_LOG_EXCEPTION_MESSAGE]}...'
```
super minor, but then also works if there is no exception messsage like `raise Exception()`

Reply:
Fixed here: https://github.com/objectiv/objectiv-analytics/pull/924/commits/ec5a938cee7e856de7273bcfcb4b124a072488d7